### PR TITLE
fix #21987: API cannot handle negative removal prices

### LINKED ISSUE #21987: Plugin: SmallSceneryObject.removalPrice is incorrect for objects that refund on removal
### Operating System

Windows 10 x64

### OpenRCT2 build

OpenRCT2, v0.4.11 (18d2b5e) provided by GitHub

### Base game

RollerCoaster Tycoon 2, RollerCoaster Tycoon Classic

### Area(s) with this issue?

This is a development issue

### Describe the issue

Scenery objects with a negative removalPrice seem to return a wrong value when accessed through the plugin API, specifically they return positive values some power of two larger than the true value, almost like there is a signedness bug somewhere. For most objects this is 256, for other things like the most expensive thing I could find in a hurry (Mythological theming golden fleece) the error is 4096.

There's every chance that this is somehow user error as well, I'm very new to JS/TS.

### Steps to reproduce

The code in question:
```ts
export function removalPriceBug(): void
{
    for (let y = 0; y < map.size.y; y++) {
        for (let x = 0; x < map.size.x; x++) {
            let tile = map.getTile(x, y);
            for (let i = 0; i < tile.numElements; i++) {
                let element = tile.getElement(i);
                if (element.type === 'small_scenery') {
                    let scenery = objectManager.getObject("small_scenery", element.object);
                    if (scenery.removalPrice != 0)
                    {
                        park.postMessage("item " + scenery.identifier + " has removal price: " + scenery.removalPrice);
                    }
                }
            }
        }
    }
}
```
This works fine for things with a positive removal cost.

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by Sadret**
Changes the return type of the getter from uint8 to money64 (=int64), which is the type that the actual property has.
Applies the same to `price`, which couldn't handle prices above ~255~ 25.50 GBP before.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
> Applies the same to price, which couldn't handle prices above 255 GBP before.

If that is true, then the value in question is not a money64. money8/money16/money32/money64 are all in tens of pence, so either the limit was 25.50 GBP, or the value is not a proper money64. I’d like to investigate this.

**Comment by Sadret** [comment]
> > Applies the same to price, which couldn't handle prices above 255 GBP before.
> 
> If that is true, then the value in question is not a money64. money8/money16/money32/money64 are all in tens of pence, so either the limit was 25.50 GBP, or the value is not a proper money64. I’d like to investigate this.

My bad, I meant 255 money units, i.e 25.50 GBP.

File: distribution/changelog.txt [hunk 1]
```diff
@@ -30,6 +30,7 @@
 - Fix: [#13294] Map corners are cut off in some directions (original bug).
 - Fix: [#14630] Non-ASCII thousands and decimal separators not processed correctly.
 - Fix: [#21974] No reason specified when attempting to place benches, lamps, or bins on path with no unconnected edges (original bug).
+- Fix: [#21987] [Plugin] API can now handle negative removal prices.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:33]: Always list the problem, not the fix.

Comment (Gymnasiast) [reply distribution/changelog.txt:33]:
```suggestion
- Fix: [#21987] [Plugin] API cannot handle negative removal prices.
```

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**